js/gdax.js

```diff
@@ -269,20 +276,39 @@ module.exports = class gdax extends Exchange {
                 'currency': market['quote'],
             };
         }
+        let type = undefined;
+        if ('liquidity' in trade) {
+            type = (trade['liquidity'] == 'T') ? 'Taker' : 'Maker';
```

Comment:
Shouldn't `type` be 'market' or 'limit' rather than 'Taker' or 'Maker'? See [kraken](https://github.com/ccxt/ccxt/blob/master/js/kraken.js#L478) by comparison.

My limit orders had 'M' in the `liquidity` field.

Reply:
type should be `market` or `limit` indeed, this is a common pitfall )

Reply:
Can the type be determined [at all](https://github.com/ccxt/ccxt/pull/1657#issuecomment-361897720)?

Among the trades I've checked in my account (about 20), when `info.liquidity` was 'M', `info.fee` was always zero, so I assume it was a limit order. When liquidity was 'T', the fee was non-zero.

Reply:
> Can the type be determined at all?

I don't think so, therefore I've set it to undefined.

> Among the trades I've checked in my account (about 20), when info.liquidity was 'M', info.fee was always zero, so I assume it was a limit order. 

Before you assumed that it was a limit order indication, you also made an implicit assumption, that info.fee was always zero, however, this is not always the case with limit orders, even when you place them for a market-making price, they can take liquidity upon arrival by matching orders and that means you will have to pay fees. The same is true with many other exchanges as well.